### PR TITLE
chore(github): add Code Health issue template

### PR DESCRIPTION
**Pull request by Hzfengsy**
## Summary

- Adds a new GitHub issue template (`code_health.yml`) for non-urgent robustness concerns, technical debt, and code quality improvements
- Creates the `code health` label for categorizing these issues
- Fills a gap between bug reports (broken now) and feature requests (new capability) for issues like potential edge-case failures or naming collisions

## Test plan

- [ ] Verify template renders correctly in GitHub issue creation UI
- [ ] Verify `code health` label is applied automatically

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/code_health.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    id: category
+    attributes:
+      label: Category
+      description: What kind of code health issue is this?
+      options:
+        - Robustness (potential edge-case failure)
+        - Technical Debt (cleanup, refactor)
+        - Naming / Consistency
+        - Missing Validation
+        - Dead Code / Unused Dependencies
```
[the remark this answers is not in the document]
Comment (Hzfengsy) [reply .github/ISSUE_TEMPLATE/code_health.yml:21]: Skipping — the current categories are sufficient for now. Testing concerns can fall under "Technical Debt" or "Robustness". We can revisit if usage shows a need.

File: .github/ISSUE_TEMPLATE/code_health.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        - Parser
+        - Printer
+        - Python Bindings
+        - Codegen
+        - Backend
+        - Build System
+        - Tests
+        - Other (please specify in description)
+    validations:
+      required: true
```
[the remark this answers is not in the document]
Comment (Hzfengsy) [reply .github/ISSUE_TEMPLATE/code_health.yml:42]: Keeping as single-select dropdown for consistency with other issue templates (bug_report, pass_bug, etc.). Users can note additional components in the description field.
